### PR TITLE
Support the creation of initial user

### PR DESCRIPTION
**Pull request by alwx**
Support for initial user creation (https://github.com/RasaHQ/rasa-x/pull/1858)

### CODE REVIEW

File: charts/rasa-x/templates/rasa-secret.yaml [hunk 1]
```diff
@@ -11,4 +11,6 @@ data:
   rasaXToken: {{ .Values.rasax.token | b64enc | quote }}
   passwordSalt: {{ .Values.rasax.passwordSalt | b64enc | quote }}
   jwtSecret: {{ .Values.rasax.jwtSecret | b64enc | quote }}
+  initialUserName: {{ .Values.rasax.initialUser.username }}
+  initialPassword: {{ .Values.rasax.initialUser.password }}
```
Comment (alwx) [inline charts/rasa-x/templates/rasa-secret.yaml]: Maybe the naming is not ideal — open to suggestions :)

File: README.md [hunk 2]
```diff
@@ -43,6 +43,8 @@ recommend to set at least these values:
 | rasax.passwordSalt                   | Password salt which Rasa X uses for the user passwords.                                    | `passwordSalt`     |
 | rasax.token                          | Token which the Rasa X pod uses to authenticate requests from other pods.                  | `rasaXToken`       |
 | rasax.jwtSecret                      | Secret which is used to sign JWT tokens of Rasa X users.                           | `jwtSecret`        |
+| rasax.initialUser.username           | A name of the user that will be created immediately after the first launch (`rasax.initialUser.password` should be specified) | `admin`            |
```
Comment (wochinge) [inline README.md]: please mention that this is only considered for Rasa X EE

File: README.md [hunk 3]
```diff
@@ -43,6 +43,8 @@ recommend to set at least these values:
...
+| rasax.initialUser.password           | Password for the initial user (leave it empty if you don't want user to be created).       | `""`               |
```
Comment (wochinge) [inline README.md]:
```suggestion
| rasax.initialUser.password           | Password for the initial user (leave it empty if you don't want the user to be created).       | `""`               |
```

File: charts/rasa-x/templates/rasa-secret.yaml (hunk 1 above)
Comment (wochinge) [reply charts/rasa-x/templates/rasa-secret.yaml]: I actually like it :-D

File: charts/rasa-x/templates/rasa-x-deployment.yaml [hunk 4]
```diff
@@ -90,6 +90,16 @@ spec:
             secretKeyRef:
               name: {{ template "rasa-x.secret" . }}
               key: "rasaXToken"
+        - name: "RASA_X_USERNAME"
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "rasa-x.secret" $ }}
```
Comment (wochinge) [inline charts/rasa-x/templates/rasa-x-deployment.yaml]:
```suggestion
              name: {{ template "rasa-x.secret" . }}
```

File: charts/rasa-x/templates/rasa-x-deployment.yaml [hunk 5]
```diff
@@ -90,6 +90,16 @@ spec:
...
+              key: "initialUserName"
+        - name: "RASA_X_PASSWORD"
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "rasa-x.secret" $ }}
```
Comment (wochinge) [inline charts/rasa-x/templates/rasa-x-deployment.yaml]:
```suggestion
              name: {{ template "rasa-x.secret" . }}
```

File: charts/rasa-x/values.yaml [hunk 6]
```diff
@@ -20,6 +20,11 @@ rasax:
   databaseName: "rasa"
   # disableTelemetry permanently disables telemetry
   disableTelemetry: false
+  initialUser:
```
Comment (wochinge) [inline charts/rasa-x/values.yaml]:
```suggestion
  # initialUser is the user which is created upon the initial start of Rasa X
  initialUser:
```

File: charts/rasa-x/templates/rasa-secret.yaml (hunk 1 above)
Comment (wochinge) [inline charts/rasa-x/templates/rasa-secret.yaml]:
```suggestion
  initialPassword: {{ .Values.rasax.initialUser.password | b64enc | quote }}
```

File: charts/rasa-x/templates/rasa-secret.yaml [hunk 7]
```diff
@@ -11,4 +11,6 @@ data:
   rasaXToken: {{ .Values.rasax.token | b64enc | quote }}
   passwordSalt: {{ .Values.rasax.passwordSalt | b64enc | quote }}
   jwtSecret: {{ .Values.rasax.jwtSecret | b64enc | quote }}
+  initialUserName: {{ .Values.rasax.initialUser.username }}
```
Comment (wochinge) [inline charts/rasa-x/templates/rasa-secret.yaml]: would you actually put this in the secret? I'm not sure 🤔

Comment (alwx) [reply charts/rasa-x/templates/rasa-secret.yaml]: yes! but it has to be base64-encoded as well

File: charts/rasa-x/templates/rasa-secret.yaml [hunk 8]
```diff
@@ -11,4 +11,6 @@ data:
   rasaXToken: {{ .Values.rasax.token | b64enc | quote }}
   passwordSalt: {{ .Values.rasax.passwordSalt | b64enc | quote }}
   jwtSecret: {{ .Values.rasax.jwtSecret | b64enc | quote }}
+  initialUserName: {{ .Values.rasax.initialUser.username | b64enc | quote }}
```
Comment (wochinge) [inline charts/rasa-x/templates/rasa-secret.yaml]: but why would we put it in the secret? Other charts also don't do this (e.g. https://github.com/helm/charts/blob/master/stable/postgresql/templates/secrets.yaml)

File: charts/rasa-x/values.yaml [hunk 9]
```diff
@@ -20,6 +20,12 @@ rasax:
   databaseName: "rasa"
   # disableTelemetry permanently disables telemetry
   disableTelemetry: false
+  # initialUser is the user which is created upon the initial start of Rasa X
+  initialUser:
+    # username specifies a name of this user
+    username: "admin"
```
Comment (wochinge) [inline charts/rasa-x/values.yaml]: how about having the default community user here?
```suggestion
    username: "me"
```

File: charts/rasa-x/templates/NOTES.txt [hunk 10]
```diff
@@ -1,5 +1,6 @@
 Thanks for installing Rasa X {{ .Values.rasax.tag }} !
 
+{{- if not .Values.rasax.initialUser.password }}
```
Comment (wochinge) [inline charts/rasa-x/templates/NOTES.txt]: slick idea! 💯

File: charts/rasa-x/values.yaml (hunk 9 above)
Comment (alwx) [reply charts/rasa-x/values.yaml]: 👍

File: charts/rasa-x/templates/rasa-secret.yaml (hunk 8 above)
Comment (alwx) [reply charts/rasa-x/templates/rasa-secret.yaml]: the official guide says that it's fine to put username and password there (https://kubernetes.io/docs/concepts/configuration/secret/), so I am not sure what is the best practice. I will investigate and update you :)

Comment (alwx) [reply charts/rasa-x/templates/rasa-secret.yaml]: You were right — it's better to remove it from there. I did it and pushed — so please check it again.

File: README.md [hunk 11]
```diff
@@ -43,6 +43,8 @@ recommend to set at least these values:
 | rasax.passwordSalt                   | Password salt which Rasa X uses for the user passwords.                                    | `passwordSalt`     |
 | rasax.token                          | Token which the Rasa X pod uses to authenticate requests from other pods.                  | `rasaXToken`       |
 | rasax.jwtSecret                      | Secret which is used to sign JWT tokens of Rasa X users.                           | `jwtSecret`        |
+| rasax.initialUser.username           | **Only for Rasa Enterprise**. A name of the user that will be created immediately after the first launch (`rasax.initialUser.password` should be specified) | `admin`            |
+| rasax.initialUser.password           | **Only for Rasa Enterprise**. Password for the initial user (leave it empty if you don't want the user to be created).       | `""`               |
```
Comment (wochinge) [inline README.md]: this one is not for enterprise only

File: charts/rasa-x/templates/rasa-deployments.yaml [hunk 12]
```diff
@@ -70,6 +70,13 @@ spec:
             secretKeyRef:
               name: {{ template "rasa-x.secret" $ }}
               key: "rasaToken"
+        - name: "RASA_X_USERNAME"
```
Comment (wochinge) [inline charts/rasa-x/templates/rasa-deployments.yaml]: we don't need in here. having it in the `rasa-x-deployment.yaml` is enough

File: charts/rasa-x/templates/rasa-x-deployment.yaml [hunk 13]
```diff
@@ -90,6 +90,16 @@ spec:
             secretKeyRef:
               name: {{ template "rasa-x.secret" . }}
               key: "rasaXToken"
+        - name: "RASA_X_USERNAME"
```
Comment (wochinge) [inline charts/rasa-x/templates/rasa-x-deployment.yaml]: that's now from the secret anymore, but from the values file.

**Comment by alwx** [comment]
@wochinge sorry, it took a while. Can you please check it again?

File: charts/rasa-x/templates/rasa-x-deployment.yaml [hunk 14]
```diff
@@ -90,6 +90,13 @@ spec:
...
+          value: "{{ $.Values.rasax.initialUser.username }}"
```
Comment (wochinge) [inline charts/rasa-x/templates/rasa-x-deployment.yaml]: should work like this, no?
```suggestion
          value: "{{ .Values.rasax.initialUser.username }}"
```

Comment (alwx) [reply charts/rasa-x/templates/rasa-x-deployment.yaml]: Hm, it worked with $ as well. But will change it

**Comment by wochinge** [comment]
can you please bump the rasa-x and rasa versions to 1.7.0 and 0.25.0 and then merge? 🎉

**Pull request merged**